docs/extensibility/extending-blocks.md

```diff
@@ -0,0 +1,129 @@
+
+## Modifying Blocks
+
+To modify the behavior of existing blocks, Gutenberg exposes the following Filters:
+
+#### `blocks.registerBlockType`
+
+Used to filter the block settings. It receives the block settings and the name of the block the registered block as arguments.
+
+#### `blocks.BlockEdit`
+
+Used to modify the block's `edit` component. It receives the original block `edit` component and returns a new wrapped component.
+
+#### `blocks.getSaveElement`
+
+A filter that applies to the result of a block's `save` function. This filter is used to replace or extend the element, for example using `wp.element.cloneElement` to modify the element's props or replace its children, or returning an entirely new element.
+
+#### `blocks.getSaveContent.extraProps`
+ 
+A filter that applies to all blocks returning a WP Element in the `save` function. This filter is used to add extra props to the root element of the `save` function. For example: to add a className, an id, or any valid prop for this element. It receives the current props of the `save` element, the block type and the block attributes as arguments.
+
+_Example:_
+
+Adding a background by default to all blocks.
+
+```js
+function addBackgroundColorStyle( props ) {
+	return Object.assign( props, { style: { backgroundColor: 'red' } } );
+}
+
+wp.hooks.addFilter(
+	'blocks.getSaveContent.extraProps',
+	'my-plugin/add-background-color-style',
+	addBackgroundColorStyle
+);
+```
+
+_Note:_ This filter must always be run on every page load, and not in your browser's developer tools console. Otherwise, a [block validation](https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/#validation) error will occur the next time the post is edited. This is due to the fact that block validation occurs by verifying that the saved output matches what is stored in the post's content during editor initialization. So, if this filter does not exist when the editor loads, the block will be marked as invalid.
+
+#### `blocks.getBlockDefaultClassName`
+
+Generated HTML classes for blocks follow the `wp-block-{name}` nomenclature. This filter allows to provide an alternative class name.
+
+_Example:_
+
+```js
+// Our filter function
+function setBlockCustomClassName( className, blockName ) {
+	return blockName === 'core/code' ?
+		'my-plugin-code' :
+		className;
+}
+
+// Adding the filter
+wp.hooks.addFilter(
+	'blocks.getBlockDefaultClassName',
+	'my-plugin/set-block-custom-class-name',
+	setBlockCustomClassName
+);
+```
+
+#### `blocks.isUnmodifiedDefaultBlock.attributes`
+
+Used internally by the default block (paragraph) to exclude the attributes from the check if the block was modified.
+
+#### `blocks.switchToBlockType.transformedBlock`
+
+Used to filters an individual transform result from block transformation. All of the original blocks are passed, since transformations are many-to-many, not one-to-one.
```

Comment:
@aduth, can you confirm this is correct? It looks like you added this hook. Do you think it makes sense to add an example?

Reply:
Yeah, reading my original code comment, it could be clearer. This filter is called on each block result of the transform. The "all of the original blocks" note is more a consideration for the filterer if they care about where the transform had been sourced from.